### DISCUSSION TITLE
How can I contribute to this repo?

### DISCUSSION THREAD

**Discussion by BrianBohe** [coin-or/Cbc#307] [Q&A]
Hello,

I would like to start contributing to this repository. Is there a beginner user guide? I didn't see tags on existing issues that could help me know if that issue is good for beginners. Do you have an internal tagged copy of the issues?  Thanks!

**Comment by h-g-s** [comment]
Hello @BrianBohe ,

Good to hear that. The first point is that CBC is composed by several projects:
https://github.com/coin-or/CoinUtils  (read files, store matrixes)
https://github.com/coin-or/Osi   (open solver interface)
https://github.com/coin-or/Cgl   (cut generation library, pre-processing, cuts)
https://github.com/coin-or/Clp  (the LP solver)

CoinUtils, for example, has no major dependencies.    Clp depends on Osi and Osi depends on CoinUtils, for example.

One part that we are now trying to improve, for example, that is not very complex, for example, is reading and saving special ordered sets.  The code to read LP and MPS files is in coinutils.
